File: employeeManagement/frontend/src/services/EmployeeService.js
import axios from 'axios';

const BASE = process.env.REACT_APP_EMPLOYEE_API_BASE_URL;

const EmployeeService = {
    getAll:       ()           => axios.get(BASE),
    getById:      (id)         => axios.get(`${BASE}/${id}`),
    create:       (employee)   => axios.post(BASE, employee),
    update:       (id, employee) => axios.put(`${BASE}/${id}`, employee),
    remove:       (id)         => axios.delete(`${BASE}/${id}`),
};

export default EmployeeService;

File: employeeManagement/backend/src/main/java/com/avishkar/service/EmployeeService.java
package com.avishkar.service;

import com.avishkar.exception.ResourceNotFoundException;
import com.avishkar.model.Employee;
import com.avishkar.repository.EmployeeRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@Transactional
public class EmployeeService {

    private final EmployeeRepository employeeRepository;

    public EmployeeService(EmployeeRepository employeeRepository) {
        this.employeeRepository = employeeRepository;
    }

    @Transactional(readOnly = true)
    public List<Employee> getAll() {
        return employeeRepository.findAll();
    }

    @Transactional(readOnly = true)
    public Employee getById(Long id) {
        return employeeRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Employee not found with id: " + id));
    }

    public Employee create(Employee employee) {
        return employeeRepository.save(employee);
    }

    public Employee update(Long id, Employee details) {
        Employee employee = getById(id);
        employee.setFirstName(details.getFirstName());
        employee.setLastName(details.getLastName());
        employee.setEmailId(details.getEmailId());
        employee.setDepartment(details.getDepartment());
        return employeeRepository.save(employee);
    }

    public void delete(Long id) {
        Employee employee = getById(id);
        employeeRepository.delete(employee);
    }
}
